### ISSUE TITLE
iOS Teams App Propagation and weird Android Behavior

### ISSUE THREAD

**Issue by gsogol** [MicrosoftDocs/msteams-docs#7902]
I was asked by internal Microsoft Teams Engineering PM to open an issue here in hopes of getting to the bottom. Thnk you in advance.

**What we did**

-     We developed a custom teams app using fluent ui, react.js components using most from https://www.npmjs.com/package/@fluentui/react-components and some from https://www.npmjs.com/package/@fluentui/react-northstar.
-     The teams app was not deployed to Microsoft Store as we're in the early stages of development. What we do instead is create a setup policy, upload the app and watch the users get the app installed as it propagates. We followed the documentation from https://learn.microsoft.com/en-us/microsoftteams/teams-app-setup-policies
-     The app uses the bot framework and [proactive messages](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/send-proactive-messages?tabs=dotnet) to start a conversation
-     Upon installation and happy path scenario, when the app is first installed, it sends a welcome message. Users see it in the native Teams Chat Notification as well as under our app. If they click on the chat, they're taken to our app. This works fine for most users
-     On a daily basis, the user receives 2 proactive messages from us (basically something for them to work on). We work with various companies to deliver bite-sized business objectives, hence, the daily messages. Each proactive message sends an adaptive card with a button that they click on to commit to each daily activity. Note: I can add you to our demo environment if you want to try it out. No information is needed besides your email but the app does require certain permissions within the tenant to operate properly. Just let me know if you want to try and replicate. I can remove once we're done troubleshooting.

**What the issues are**

We have 2 issues that we're dealing with:

-     When we create the aforementioned linked setup policy above, it actually works quite well and both windows and mac users see the app installed over time on their desktop Teams app. Great! However, iOS users (iPhone and iPad) never do. At first we thought more time was needed but nothing ever happened. We then tried it on a completely different tenant. We have an ex Microsoft employee working with us and he had a tenant he's using for himself for many years now. He tried it there as well and exactly the same issue occurs. As you can imagine, we're at a point where we tell companies, sorry, this only works for desktop users. Android users seem ok with this issue. There are no screenshots to provide as the issue is about propagation of the app itself.
-     The second issue has to do with Android users. Now, they do get the app installed and they do get proactive messages sent to them. As I mentioned above, upon receiving a proactive message in the Teams native Chat tab on Android, a completely new behavior appears. For all others, upon clicking on our message, they're taken to our app. On Android, though, they see 2 tabs within the native Chat tab on Teams. When clicking on More, an error occurs. We have no idea why this appeared. We never designed it, as our application has our own tabs and nothing to do with More. See below.
Screenshot can be seen [here](https://drive.google.com/file/d/1aEz6VYnwqRShLp0Eit4r7CE5QukovGj5/view?usp=share_link)

**What we tried**

    Contacted Support. After 3 weeks, we were told sorry, we do not support custom teams development. We said, the issue is not with our app, the issues are propagation and a weird Android tab. They told us nothing they can do and closed the cases.
    We are at a mercy of your team and hope we can figure this out.

**Comment by ChetanSharma-msft** [comment]
Hello @gsogol - Thanks for reporting your issue.
We will investigate your issue and let you know the updates.

**Comment by sagar-malpure** [comment]
@gsogol, Could you please add "adpraveen@yfqxx.onmicrosoft.com" to your demo environment to repro the issue.

**Comment by gsogol** [comment]
Yes, added. 
1) The person will receive 2 proactive messages daily to test the Android issue we are seeing. That's the one where a mysterious More tab shows up and upon clicking it, gives us an error. Clicking on the chat message, also does not take us to the app. 
2) The other issue with iOS where the app does not get installed during propagation via setup policy, you can try that yourself in your tenant. I am wondering if somehow something needs to be enabled on your end. This talks about connecting some Apple ID 

https://learn.microsoft.com/en-us/microsoftteams/platform/concepts/deploy-and-publish/appsource/prepare/update-apple-store-team-connect-id 

but I'm unsure. We are not publishing to the teams store. We are uploading manually to each org. But feels like something was not enabled or some missing step that is not part of docs

**Comment by gsogol** [comment]
The app zip file can be found [here](https://drive.google.com/file/d/1tY_hz5CxMMq7a467wEaYpR9Vnl9JU-eF/view?usp=share_link)

**Comment by SubbaReddi** [comment]
@gsogol: 
1. "More" Tab apps are showing error "something went wrong" in mobile. We see it as product bug and reported the same to product group.
2. We see it as issue with values given in manifest.
Observed issue with bot commands. Commands are given empty in manifest. If we remove commandLists section from manifest, app is showing in iOS.
"commandLists": [
{
"scopes": ["personal", "team", "groupchat"],
"commands": []
}
]
So, if commands are not required, please remove the respective section from manifest to fix the issue. 

Hope it helps?

**Comment by gsogol** [comment]
Thank you so much. 

1) so you will let us know when this is fixed?
2) we will try this tomorrow and report back! 🙌

**Comment by gsogol** [comment]
**Update**. The iOS propagation now works when removing the empty commandLists from manifest. 

**Remaining issue** is the mysterious More tab and a proactive message click within the native Chat does not take you to the app unlike on other devices.

**Comment by gsogol** [comment]
@SubbaReddi @Nivedipa-MSFT @ChetanSharma-msft   could you confirm that the Android issue is being looked at? Forgive my uneasiness, we need to roll it out to many users and many users Android so we just want to know that it's being looked at. Thank you as always.

**Comment by SubbaReddi** [comment]
@gsogol : In Teams tab/bot application we observed extra 'More' tab and on click of it gives us an error. "Something went wrong". If you are referring to the same issue, we reported as bug to product team. And it may take time to get the fix ETA.

**Comment by ChetanSharma-msft** [comment]
Hello @gsogol - Currently we don't have exact ETA to share but we are actively checking the updates with engineering team and let you know once it is resolved.

**Comment by sagar-malpure** [comment]
Hello @gsogol, we haven't received any proactive messages after installing your app manifest so could not repro the below issue. Could you please help us to repro the same.

![image](https://user-images.githubusercontent.com/85691087/212872804-9a2024dd-cab8-4451-820e-ecd648e0521d.png)

**Comment by gsogol** [comment]
Yes of course. Could you provide me your email addresses? Assuming we are testing the Android issue

**Comment by sagar-malpure** [comment]
[adpraveen@yfqxx.onmicrosoft.com](mailto:adpraveen@yfqxx.onmicrosoft.com) this is my test user id.

**Comment by gsogol** [comment]
@sagar-malpure  we already added this user over the weekend. The idea is to validate via the Android phone using the app. We see you're using and clicking via chrome browser.

**Comment by gsogol** [comment]
@sagar-malpure just making sure you were able to recreate the issue and getting what you need. If a call is easier, we can do that too to speed up the process.

**Comment by v-nikhilsoma** [comment]
@gsogol Could you please let us know your availability? so that we can schedule a meeting to look into this further.

**Comment by gsogol** [comment]
Any day, any time from 7am to midnight est timezone. jeff@prohabits.com
